nailgun/entity_mixins.py

```diff
@@ -299,11 +311,14 @@ def __init__(self, server_config=None, **kwargs):
         for field_name, field_value in kwargs.items():  # e.g. ('admin', True)
             field = self._fields[field_name]  # e.g. A BooleanField object
             if isinstance(field, OneToOneField):
-                setattr(self, field_name, _make_entity_from_id(
-                    field.gen_value(),
-                    field_value,
-                    self._server_config
-                ))
+                if field_value is None:
+                    setattr(self, field_name, field_value)
```

Comment:
:smile:

This definitely seems like more sane behaviour. I don't know what I was smoking when I hammered out the original design.

Actually, I *do* know what I was smoking. I was smoking the "I'm-not-unit-testing-everying" pipe. The process of expanding unit test coverage is what really forced me to redesign this. This is yet another argument in favor of developers writing tests as part of  their workflow.